packages/@react-spectrum/table/src/index.ts

```diff
@@ -13,8 +13,13 @@
 /// <reference types="css-module-types" />
 
 export {TableView} from './TableView';
-import {Column} from '@react-stately/table';
+import {Column, TableSection} from '@react-stately/table';
 import {SpectrumColumnProps} from '@react-types/table';
+import {SpectrumTableSectionProps} from './TableView';
+
+// Override TS for TableSection to require title prop.
+const SpectrumTableSection = TableSection as <T>(props: SpectrumTableSectionProps<T>) => JSX.Element;
+export {SpectrumTableSection as TableSection};
```

Comment:
Note we are exporting a TableSection and not reusing the Section collection. This feels like it makes more sense since all of Table's collection components are table specific anyways